BlueDBOnDisk/src/main/java/org/bluedb/disk/collection/ReadableCollectionOnDisk.java

```diff
@@ -118,17 +118,19 @@ protected void ensureCorrectKeyTypes(Collection<BlueKey> keys) throws BlueDbExce
 	}
 
 	protected static SegmentSizeSetting determineSegmentSize(ReadableCollectionMetadata metaData, Class<? extends BlueKey> keyType, SegmentSizeSetting requestedSegmentSize, boolean isNewCollection) throws BlueDbException {
-		SegmentSizeSetting existingSegmentSize = metaData.getSegmentSize();
-		if (existingSegmentSize == null) {
+		SegmentSizeSetting segmentSize = metaData.getSegmentSize();
+		if (segmentSize == null) {
 			if (!isNewCollection) {
-				return SegmentSizeSetting.getOriginalDefaultSettingsFor(keyType);

```

Comment:
Was there a reason that you weren't creating the meta data file when we initialize an existing collection that doesn't already have one?